src/Adapter/Module/AdminModuleDataProvider.php

```diff
@@ -611,15 +453,6 @@ protected function fallbackOnCatalogCache()
         return ($this->catalog_categories && $this->catalog_modules);
     }
 
-    protected function fallbackOnManageCache()
-    {
-        // Fallback on data from cache if exists
-        $this->manage_categories = $this->getModuleCache(self::_CACHEFILE_INSTALLED_CATEGORIES_, false);
-        $this->manage_modules    = $this->getModuleCache(self::_CACHEFILE_INSTALLED_MODULES_, false);
-
-        return ($this->manage_categories && $this->manage_modules);
-    }
-
     private function getModuleCache($file, $check_freshness = true)
     {
```

Comment:
I need more time to review this file ... I think too much responsabilities for only one class, did you take a look at the Theme implementation with Manager and Repository ?